monzo/monzo.py

```diff
@@ -276,3 +277,57 @@ def update_transaction_notes(self, transaction_id, notes):
            :rtype: The updated transaction object.
         """
         return self.update_transaction_metadata(transaction_id, 'notes', notes)
+
+    def create_receipt(self, transaction_id, total, currency, items, taxes=None, payments=None, merchant=None):
+        """Create a receipt for a given transaction. (https://docs.monzo.com/#create-receipt) (https://docs.monzo.com/#receipts)
+           :param transaction_id: The unique identifier for the transaction for which the receipt should be added to.
+           :param total: The integer amount for the value of the transaction.
+           :param currency: The str of the currency used for the transaction; usually `GBP`.
+           :param items: The list of items detailing the products included in the total. (https://docs.monzo.com/#receipt-items)
+           :param taxes: Not required by default: The list of taxes (e.g. VAT) added onto the total. (https://docs.monzo.com/#receipt-taxes)
+           :param payments: Not required by default: A list of payments, indicating how the customer paid the total. (https://docs.monzo.com/#receipt-payments)
+           :param merchant: Not required by default: Extra information about the merchant you shopped at. (https://docs.monzo.com/#receipt-merchant)
+           :rtype: A tuple object containing the receipt object (receipt_id and infoformation) you "PUT", repeated back to you & the generate external_id.
+        """
+        url = "{0}/transaction-receipts".format(self.API_URL)
+        # Generate a random external_id consisting of 8 digits. 
+        external_id = "Order-" + str(random.randint(10000000, 99999999))
+        # Init base receipt structure with required properties.
+        data = {
+            "transaction_id": transaction_id,
+            "external_id": external_id,
+            "total": total,
+            "currency": currency,
+            "items": items
+        }
+        # Check for non-required properties and are the correct type (taxes, payments & merchant).
+        if (taxes is not None) and (type(taxes) is list):
+            data["taxes"] = taxes
+        if (payments is not None) and (type(payments) is list):
+            data["payments"] = payments
+        if merchant is not None and (type(merchant) is dict):
+            data["merchant"] = merchant
+        response = self.oauth_session.make_request(url, data=json.dumps(data), method='PUT')
```

Comment:
Yes, now this I wanted to ask you about. Usually I would just use something along the lines in requests of `resp = requests.put(url, json=data) however, when I attempted to do this with the `make_request()` I was running into issues.

Reply:
I'd imagine the problem is requests doesn't like http requests with both the data and json parameters set as it doesn't know which payload to take. Try editing MonzoOAuth2Client so that it doesn't use the data parameter.

If that works then we can think about whether it's worth making this change.

Reply:
I believe passing `response = self.oauth_session.make_request(url, json=receipt, method='PUT')` is working correctly, however, I am currently getting a 403 `insufficient permissions` error to the endpoint when attempting to create a receipt.

Reply:
I can successfully create a receipt now. However, the new issue is that for some reason as per the documentation the API should return the `receipt_id and the rest of the information you provided, repeated back to you.` however, no data in being provided in the response. I am only recieveing `b''` when reviewing `response.content & response.text`. Therefore, the overall request fails as there is no json data to decode.

Must be an issue with the API on Monzo's end?

Reply:
I'm trying to replicate this but I'm getting 403 errors when creating a receipt. What did you do to fix this?

Reply:
Ah, I just needed to try a different transaction

Reply:
Yep, I get the same lack of response. I wouldn't worry too much about it as the receipt is still created.

I noticed that we still get 403 errors when trying to delete a receipt but based on the forums we're not alone in this. Hopefully this will be fixed soon as this would need to be fixed before we can merge.

Reply:
Exactly what I have found. I have been discussing with other developers on Slack and they are having the same issues. Either a documentation or API issue. I will be submitting a post on the Monzo forums shortly.

Reply:
Yeah I originally tried it on the first transaction which was me sending a little bit of cash from a other account. When I tried on a proper purchase then it worked.